FRAuth/FRAuth/Callbacks/DeviceProfileCallback.swift

```diff
@@ -26,18 +30,27 @@ import Foundation
     /// - Parameter json: JSON object of HiddenValueCallback
     /// - Throws: AuthError.invalidCallbackResponse for invalid callback response
     required init(json: [String : Any]) throws {
-        try super.init(json: json)
         
-        // For now, force Callback.type as DeviceAttributeCallback
-        type = "DeviceAttributeCallback"
+        guard let outputs = json["output"] as? [[String: Any]] else {
+                throw AuthError.invalidCallbackResponse(String(describing: json))
+        }
         
-        if let idUrlString = id, let idUrl = URLComponents(string: idUrlString), let queryItems = idUrl.queryItems {
-            for queryItem in queryItems {
-                if queryItem.name == "attributes", let queryValue = queryItem.value {
-                    attributes.append(queryValue)
-                }
+        for output in outputs {
+            if let outputName = output["name"] as? String, outputName == "location", let outputValue = output["value"] as? Bool {
+                locationRequired = outputValue
+            }
+            else if let outputName = output["name"] as? String, outputName == "metadata", let outputValue = output["value"] as? Bool {
+                metadataRequired = outputValue
+            }
+            else if let outputName = output["name"] as? String, outputName == "message", let outputValue = output["value"] as? String {
+                message = outputValue
             }
         }
+        
+        try super.init(json: json)
+        
+        // For now, force Callback.type as DeviceAttributeCallback
+        type = "DeviceAttributeCallback"
```

Comment:
Should the type be updated to DeviceProfileCallback?

Reply:
Corrected with the latest commit, thanks for pointing this out.